server/src/test/java/io/crate/planner/operators/JoinTest.java

```diff
@@ -768,4 +769,84 @@ public void test_correlated_subquery_can_access_all_tables_from_outer_query_that
             """
         );
     }
+
+    @Test
+    public void test_nested_join_pair_ordering() throws Exception {
+        QueriedSelectRelation relation = e.analyze(
+            """
+                SELECT *
+                FROM
+                t1
+                  JOIN (t2 JOIN t3 ON t2.i = t3.z)
+                  ON t1.i = t2.i
+                """
+        );
+        assertThat(relation.joinPairs().get(0)).hasPairs(T3.T2, T3.T3);
+        assertThat(relation.joinPairs().get(1)).hasPairs(T3.T1, T3.T2);
+    }
+
+    @Test
+    public void test_nested_joins_with_sub_query_join_pair_ordering() throws Exception {
+        QueriedSelectRelation relation = e.analyze(
+            """
+                SELECT *
+                    FROM t1
+                    JOIN t2 on (t1.i = t2.i)
+                    JOIN (select 1 as foo) temp ON (temp.foo = t1.i AND t2.i = temp.foo)
+                """);
+
+        assertThat(relation.joinPairs().get(0)).hasPairs(T3.T1, T3.T2);
+        assertThat(relation.joinPairs().get(1)).hasPairs(T3.T2, relation("temp"));
+    }
+
+    @Test
+    public void test_nested_join_pg_query_join_pair_ordering() {
+        QueriedSelectRelation relation = e.analyze(
+            """
+                SELECT
```

Comment:
I suspect it popped up in some other tests because of UseJdbc(true)? Maybe can add a comment about the origin of that query - that it was some really failing scenario.

Reply:
This is jdbc driver query, which i picked up by debugging.